test/commands/init_test.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+'use strict';
+
+let assert = require('chai').assert;
+let mockery = require('mockery');
+let sinon = require('sinon');
+
+let match = sinon.match;
+
+suite('init', () => {
+  let InitCommand;
+  let sandbox,
+    originalPlatform,
+    execStub,
+    yeomanEnvStub,
+    inquirerStub,
+    elementGeneratorStub,
+    applicationGeneratorStub
+    ;
+
+  setup(() => {
+    sandbox = sinon.sandbox.create();
+    originalPlatform = Object.getOwnPropertyDescriptor(process, 'platform');
+    mockery.enable({useCleanCache: true});
+    mockery.registerAllowables(['../../lib/commands/init']);
+    mockDependencies();
+    //We need to re-import this each time mockery gets set up
+    // so that we have fresh stubs/spies
+    InitCommand = require('../../lib/commands/init').InitCommand
+  });
+
+  teardown(() => {
+    sandbox.restore();
+    Object.defineProperty(process, 'platform', originalPlatform);
+    mockery.disable();
+  });
+
+  function stubInquirer(generatorName) {
+    inquirerStub.prompt.returns(Promise.resolve({
+      generatorName: generatorName,
+    }))
+  }
+
+  test('registers default template generators', (done) => {
+    let registeredGens = {};
+    let registerSpy = sandbox.spy(function (gen, name) {
+      registeredGens[name] = gen;
+    });
+    yeomanEnvStub.prototype.registerStub = registerSpy;
+
+    yeomanEnvStub.prototype.getGeneratorsMeta.returns(registeredGens);
+    stubInquirer('polymer-init-element:app');
+
+    let init = new InitCommand();
+    let run = init.run({}, {});
+
+    return waitForPromise(run, done, () => {
```

Comment:
I don't think `waitForPromise` is necessary. You should be able to do:

```
return init.run({}, {})
  .then(() => {
    assert.isTrue(...);
  });
```

I think you need to remove the `done` parameter from the test function for mocha to wait on a Promise.